client/dive-common/components/ConfidenceFilter.vue

```diff
@@ -1,43 +1,67 @@
-<script>
+<script lang="ts">
+import { defineComponent } from '@vue/composition-api';
 import { throttle, debounce } from 'lodash';
 
-export default {
+export default defineComponent({
   name: 'ConfidenceFilter',
   props: {
     confidence: {
       type: Number,
-      required: true,
+      default: 0,
     },
-  },
-  created() {
-    this.updateConfidence = throttle(this.updateConfidence, 100);
-    this.emitEnd = debounce(this.emitEnd, 200);
-  },
-  methods: {
-    updateConfidence(value) {
-      this.$emit('update:confidence', value);
+    text: {
+      type: String,
+      default: 'Confidence Threshold',
     },
-    emitEnd() {
-      this.$emit('end');
+    min: {
+      type: Number,
+      default: 0,
     },
+    color: {
+      type: String,
+      default: null,
+    },
+  },
```

Comment:
I think this is needed for indication of type besides a large row of similar text but it seems to no longer be used:
![image](https://user-images.githubusercontent.com/61746913/135089478-5eb55480-c5d7-4fda-b99c-765ae4caeb10.png)

Reply:
I switched the circles to blocks because the blue and colored circles were making me go crosseyed.  With the blocks, I think the shape change makes it easier for the eye to tune out the blue of the sliders.

Spent an hour trying to style it (again) but it's just such a mess.